### ISSUE TITLE
java.lang.NullPointerException: null  at com.serotonin.bacnet4j.base.BACnetUtils.dottedStringToBytes (BACnetUtils.java:112)

### ISSUE THREAD

**Issue by thisisbrians** [Frozenlock/wacnet#7]
I'm getting an error upon trying to start up on a Raspberry Pi 3. I'm using the stock, pre-built 2.1.1 release and oracle-java8-jdk.

```
root@raspberrypi:/home/pi# java -version
java version "1.8.0_65"
Java(TM) SE Runtime Environment (build 1.8.0_65-b17)
Java HotSpot(TM) Client VM (build 25.65-b01, mixed mode)
```

```
root@raspberrypi:/home/pi# java -jar wacnet-2.1.1-standalone.jar

*Error*: The application encountered an error while initiating. Try deleting the configs.clj file. If the problem persists, you can contact us and include the following stacktrace :

java.lang.NullPointerException: null
 at com.serotonin.bacnet4j.base.BACnetUtils.dottedStringToBytes (BACnetUtils.java:112)
    com.serotonin.bacnet4j.npdu.ip.IpNetworkUtils.toOctetString (IpNetworkUtils.java:62)
    com.serotonin.bacnet4j.npdu.ip.IpNetwork.initialize (IpNetwork.java:220)
    com.serotonin.bacnet4j.transport.DefaultTransport.initialize (DefaultTransport.java:184)
    com.serotonin.bacnet4j.LocalDevice.initialize (LocalDevice.java:228)
    sun.reflect.NativeMethodAccessorImpl.invoke0 (NativeMethodAccessorImpl.java:-2)
    sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    java.lang.reflect.Method.invoke (Method.java:497)
    clojure.lang.Reflector.invokeMatchingMethod (Reflector.java:93)
    clojure.lang.Reflector.invokeNoArgInstanceMember (Reflector.java:313)
    bacure.local_device$initialize_BANG_$fn__22020.invoke (local_device.clj:220)
    bacure.local_device$initialize_BANG_.invokeStatic (local_device.clj:220)
    bacure.local_device$initialize_BANG_.invoke (local_device.clj:202)
    bacure.local_device$reset_local_device_BANG_.invokeStatic (local_device.clj:291)
    bacure.local_device$reset_local_device_BANG_.invoke (local_device.clj:275)
    bacure.local_device$reset_local_device_BANG_.invokeStatic (local_device.clj:285)
    bacure.local_device$reset_local_device_BANG_.invoke (local_device.clj:275)
    bacure.local_device$load_local_device_backup_BANG_.invokeStatic (local_device.clj:316)
    bacure.local_device$load_local_device_backup_BANG_.invoke (local_device.clj:311)
    bacure.core$boot_up_BANG_.invokeStatic (core.clj:13)
    bacure.core$boot_up_BANG_.invoke (core.clj:8)
    wacnet.local_device$initialize_BANG_.invokeStatic (local_device.clj:14)
    wacnet.local_device$initialize_BANG_.invoke (local_device.clj:10)
    wacnet.local_device$initialize_with_exit_on_fail_BANG_.invokeStatic (local_device.clj:59)
    wacnet.local_device$initialize_with_exit_on_fail_BANG_.invoke (local_device.clj:55)
    wacnet.server$_main.invokeStatic (server.clj:32)
    wacnet.server$_main.doInvoke (server.clj:28)
    clojure.lang.RestFn.invoke (RestFn.java:397)
    clojure.lang.AFn.applyToHelper (AFn.java:152)
    clojure.lang.RestFn.applyTo (RestFn.java:132)
    wacnet.server.main (:-1)
```

**Comment by Frozenlock** [comment]
Thanks for opening the issue.
Could you check if you have OpenVPN (or something similar) working in the background?

(See #1)

**Comment by thisisbrians** [comment]
I do indeed have OpenVPN running—and unfortunately that's how I remotely access this device. I missed that bit about OpenVPN in #1. Is there any way to get the two to play nicely together? I'm happy to hack at the source if you can point me in the right direction.

**Comment by Frozenlock** [comment]
For the time being I don't have OpenVPN on my machine, so it's a little hard to pinpoint exactly where the problem is.

However, if I had to guess I'd say that OpenVPN is somehow messing with the functions we use to determine what is the broadcast address.

[This](https://github.com/Frozenlock/bacure/blob/master/src/bacure/local_device.clj#L163) is where we get the broadcast address when initializing a local BACnet device.

If you want to dig a little, [get-any-ip](https://github.com/Frozenlock/bacure/blob/master/src/bacure/network.clj#L39) or more probably [get-broadcast-address](https://github.com/Frozenlock/bacure/blob/master/src/bacure/network.clj#L48) must be returning nil/null when used with OpenVPN. 

Both of those can be tested via the REPL. (Tho you might need to close OpenVPN before running Wacnet and *then* open it.)

**Comment by thisisbrians** [comment]
I was able to get it working by hardcoding the broadcast address as a parameter in this block: https://github.com/Frozenlock/wacnet/blob/master/src/clj/wacnet/local_device.clj#L14

Thanks for guiding me down the correct rabbit hole!

**Comment by Frozenlock** [comment]
Great!

If all that is really needed is to manually provide a broadcast address, you can do it with the config file generated by Wacnet.

**Comment by Frozenlock** [comment]
For those encountering this problem, it can be avoided with a config file providing a broadcast address.

Put a file `configs.clj` alongside Wacnet before running it.
```clj
{:broadcast-address "192.168.0.255"}
```

**Comment by Frozenlock** [comment]
Should have been fixed by 1d51597.
